### ISSUE TITLE
bug: `Generate Unit Tests` breaks the existing tests and causes errors when the file already exists

### ISSUE THREAD

**Issue by mkondratek** [sourcegraph/jetbrains#1521]
## Scenario

1. Having a test file already generated
2. Run `Generate Unit Tests` for the original file

### Actual

Cody adds file path in the beginning of generated area (middle of the file).
Cody does not close the test class brackets.
An error occurs.

### Expected

Cody correctly adds more tests cases (?).
No error.

`com.sourcegraph.cody.edit.exception.EditCreationException`

```
com.sourcegraph.cody.edit.exception.EditCreationException: Failed to create the action for: TextEdit(type=replace, range=Range(start=Position(line=19, character=0), end=Position(line=34, character=40)), position=null, value=src/test/java/MainTest.java

    @Test
    public void testAddNegativeOverflow() {
        assertEquals(Integer.MIN_VALUE, Main.add(Integer.MIN_VALUE, -1));
    }

    @Test 
    public void testAddPositiveOverflow() {
        assertEquals(Integer.MAX_VALUE, Main.add(Integer.MAX_VALUE, 1));
    }

    @Test
    public void testAddWithZero() {
        assertEquals(5, Main.add(5, 0));
        assertEquals(0, Main.add(0, 0));)
	at com.sourcegraph.cody.edit.sessions.FixupSession$performInlineEdits$1.invoke$lambda$2(FixupSession.kt:343)
	at com.intellij.openapi.command.WriteCommandAction.lambda$runWriteCommandAction$4(WriteCommandAction.java:338)
	at com.intellij.openapi.command.WriteCommandAction$BuilderImpl.lambda$doRunWriteCommandAction$1(WriteCommandAction.java:144)
	at com.intellij.openapi.application.impl.RwLockHolder.runWriteAction(RwLockHolder.kt:344)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:883)
	at com.intellij.openapi.command.WriteCommandAction$BuilderImpl.lambda$doRunWriteCommandAction$2(WriteCommandAction.java:142)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:225)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:187)
	at com.intellij.openapi.command.WriteCommandAction$BuilderImpl.doRunWriteCommandAction(WriteCommandAction.java:151)
	at com.intellij.openapi.command.WriteCommandAction$BuilderImpl.run(WriteCommandAction.java:118)
	at com.intellij.openapi.command.WriteCommandAction.runWriteCommandAction(WriteCommandAction.java:338)
	at com.intellij.openapi.command.WriteCommandAction.runWriteCommandAction(WriteCommandAction.java:326)
	at com.sourcegraph.cody.edit.sessions.FixupSession$performInlineEdits$1.invoke(FixupSession.kt:329)
	at com.sourcegraph.cody.edit.sessions.FixupSession$performInlineEdits$1.invoke(FixupSession.kt:324)
	at com.sourcegraph.cody.edit.widget.LensWidgetGroup.withListenersMuted(LensWidgetGroup.kt:110)
	at com.sourcegraph.cody.edit.sessions.FixupSession.performInlineEdits(FixupSession.kt:324)
	at com.sourcegraph.cody.edit.sessions.FixupSession.performWorkspaceEdit(FixupSession.kt:292)
	at com.sourcegraph.cody.agent.CodyAgentService$2.invoke$lambda$5(CodyAgentService.kt:84)
	at com.sourcegraph.cody.agent.CodyAgentClient.lambda$acceptOnEventThread$0(CodyAgentClient.java:126)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:209)
	at com.intellij.openapi.application.TransactionGuardImpl.access$100(TransactionGuardImpl.java:22)
	at com.intellij.openapi.application.TransactionGuardImpl$1.run(TransactionGuardImpl.java:191)
	at com.intellij.openapi.application.impl.RwLockHolder.runIntendedWriteActionOnCurrentThread(RwLockHolder.kt:204)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:830)
	at com.intellij.openapi.application.impl.ApplicationImpl$2.run(ApplicationImpl.java:419)
	at com.intellij.openapi.application.impl.RwLockHolder.runWithEnabledImplicitRead(RwLockHolder.kt:138)
	at com.intellij.openapi.application.impl.RwLockHolder.runWithImplicitRead(RwLockHolder.kt:129)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWithImplicitRead(ApplicationImpl.java:1152)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:81)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:123)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:43)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:792)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:739)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:733)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:761)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.kt:699)
	at com.intellij.ide.IdeEventQueue._dispatchEvent$lambda$12(IdeEventQueue.kt:593)
	at com.intellij.openapi.application.impl.RwLockHolder.runWithoutImplicitRead(RwLockHolder.kt:105)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.kt:593)
	at com.intellij.ide.IdeEventQueue.access$_dispatchEvent(IdeEventQueue.kt:77)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:362)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:361)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:843)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:361)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:356)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$1(IdeEventQueue.kt:1022)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:106)
	at com.intellij.ide.IdeEventQueueKt.performActivity(IdeEventQueue.kt:1022)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$7(IdeEventQueue.kt:356)
	at com.intellij.openapi.application.impl.RwLockHolder.runIntendedWriteActionOnCurrentThread(RwLockHolder.kt:209)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:830)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.kt:398)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)
Caused by: java.lang.IndexOutOfBoundsException: Wrong line: 34. Available lines count: 34
	at com.intellij.openapi.editor.impl.LineSet.checkLineIndex(LineSet.java:197)
	at com.intellij.openapi.editor.impl.LineSet.getLineStart(LineSet.java:178)
	at com.intellij.openapi.editor.impl.DocumentImpl.getLineStartOffset(DocumentImpl.java:1016)
	at com.sourcegraph.cody.edit.fixupActions.ReplaceUndoableAction.createBeforeMarker(ReplaceUndoableAction.kt:60)
	at com.sourcegraph.cody.edit.fixupActions.ReplaceUndoableAction.<init>(ReplaceUndoableAction.kt:17)
	at com.sourcegraph.cody.edit.sessions.FixupSession$performInlineEdits$1.invoke$lambda$2(FixupSession.kt:335)
	... 60 more
```

![image](https://uploads.linear.app/15b96a4d-ffc0-481a-ba9c-9d840020f7bc/8d9c9ee1-916c-4e75-80c9-9e817ae4ffb3/d737fe18-f4b4-40c7-a400-0809189578c5?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzE1Yjk2YTRkLWZmYzAtNDgxYS1iYTljLTlkODQwMDIwZjdiYy84ZDljOWVlMS05MTZjLTRlNzUtODBjOS05ZTgxN2FlNGZmYjMvZDczN2ZlMTgtZjRiNC00MGM3LWE0MDAtMDgwOTE4OTU3OGM1IiwiaWF0IjoxNzE2MzM4NjQ3LCJleHAiOjMzMjg2ODk4NjQ3fQ.fKjJEdAe1LlWKp4oEdLFyOM3bunB5vYnFrNrLC4PI00)

**Comment by pkukielka** [comment]
Looks like the same issues appears in newest VSCode dev build version.
@kalanchan @taylorsperry I'm wondering if that in any way changes priority of this task or maybe we want to ask someone from VSCode team (who knows VSCode codebase better) to look into it?

**Comment by taylorsperry** [comment]
Thanks, @pkukielka. If this is occurring in VSC as well, I think it's a P1. @abeatrix do you have any ideas about this?

**Comment by abeatrix** [comment]
> Thanks, @pkukielka. If this is occurring in VSC as well, I think it's a P1. @abeatrix do you have any ideas about this? 

I was out last week and still trying to catch up, so I don't have any ideas on top of my head right now since this is a new issue from the latest dev build. Will look into it tomorrow and report back.

**Comment by abeatrix** [comment]
I suspect https://github.com/sourcegraph/cody/pull/4172 is the cause of the issue. @pkukielka can you confirm if the commit before this PR works expectedly for you?

**Comment by odisseus** [comment]
I cannot reproduce this issue. I have seen a few different stack traces, but not this particular one.

**Comment by odisseus** [comment]
Probably fixed in 6.0.x. Feel free to reopen if this error happens again.
